src/tests/support.cpp

```diff
@@ -823,11 +800,7 @@ rnp_tests_key_search(rnp::KeyStore *keyring, const std::string &uid)
     if (!keyring || uid.empty()) {
         return NULL;
     }
-
-    pgp_key_search_t srch_userid(PGP_KEY_SEARCH_USERID);
-    strncpy(srch_userid.by.userid, uid.c_str(), sizeof(srch_userid.by.userid));
-    srch_userid.by.userid[sizeof(srch_userid.by.userid) - 1] = '\0';
-    return keyring->search(srch_userid);
+    return keyring->search(rnp::KeyUIDSearch(uid));
```

Comment:
Copilot suggests a change to improve coverage :)
```
    auto search = rnp::KeyUseridSearch(uid);
    assert(search.name() == "userid");
    assert(search.value() == uid);
    return keyring->search(search);
```

Reply:
These helper functions are designed to not assert internally - they just to make sure everything is fine on caller level.

Reply:
Ok, let me say it differently. 

There is a class factory ```std::unique_ptr<KeySearch>KeySearch::create(const pgp_key_id_t &keyid)```  and  there is  limited testing of positive scenarious for it. It is possible to add some extra checks to the helper functions to increase coverage.

Reply:
@maxirmx Got it, however I'm not sure whether we should test internal classes through internal interfaces, while 'officially' exposed is only FFI interface. I'll update FFI tests with some more edge cases.